### PR TITLE
Helm repository and automated chart build support

### PR DESCRIPTION
**Pull request by jonandernovella**
### Describe the pull request:

- [ ] Bug fix
- [x] Functional change
- [x] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

A Helm chart repository is now hosted under the branch `gh-pages`, which will contain the charts maintained under the `master` branch. A test has been developed in order to check the validity of the charts format and subsequently, automatically build and push the chart packages.
### Changes made:

1) Charts now are all localted in a new folder named `charts`
2) Whenever changes arer made to charts in the master branch, a new test build will be triggered to package new charts
3) Charts are now "publicly available" under `gh-pages`

### Mentions:

@jbygdell 
@blankdots

### CODE REVIEW

**Review by jbygdell — APPROVED** [review]

**Comment by jbygdell** [comment]
The two first travis builds are duplicates due to being run on travis-ci.com and travis-ci.org respectively.
Although the repo configurations are not shared hence the first two will always fail since they don't have any ENVs configured. They have been disabled hence forth.

File: .travis.yml [hunk 1]
```diff
@@ -1,76 +1,176 @@
...
+      # This moves Kubernetes specific config files.
+      # Inspured by https://github.com/LiliC/travis-minikube - Provided under Apache License
+        - CHANGE_MINIKUBE_NONE_USER=true
+      before_script:
+        # Install socat
+        - sudo apt update && sudo apt install socat
+        # Make root mounted as rshared to fix kube-dns issues.
+        - sudo mount --make-rshared /
+        # Download kubectl, which is a requirement for using minikube.
+        - curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/v1.13.0/bin/linux/amd64/kubectl && chmod +x kubectl && sudo mv kubectl /usr/local/bin/
```
Comment (blankdots) [inline .travis.yml:45]: here we are setting the Kubernetes version to 1.13.0, and we should be good, however we need to think of something like https://docs.travis-ci.com/user/build-matrix/ in order to test the setup on multiple (future, or even past) versions of kubernetes - this is just an idea does not need to be implemented as part of this PR.

The same idea goes for minikube version ... having a controlled version that we know works and having a separate stage for latest ..

File: .travis.yml [hunk 2]
```diff
@@ -1,76 +1,176 @@
...
+        # install LocalEGA
+        - helm install --name localega ega-charts/localega --set persistence.enabled="false",config.broker_username="$MQUSER",config.cega_endpoint="$CEGA_ENDPOINT",config.cega_mq_host="cega-mq",config.cega_username="lega",config.cega_vhost="lega",config.data_storage_url="http://minio:9000",config.postgres_user="lega",secrets.cega_creds="$CEGA_CREDS",secrets.cega_mq_pass="$CEGA_MQ_PASS",secrets.keys_password="$KEYS_PASSWORD",secrets.lega_password="$LEGA_PASSWORD",secrets.mq_password="$MQ_PASSWORD",secrets.postgres_password="$POSTGRES_PASWORD",secrets.s3_access="$MINIO_ACCESS",secrets.s3_secret="$MINIO_SECRET"
+        - sleep 180
```
Comment (blankdots) [inline .travis.yml:89]: we will need a similar preflight mechanism, that will check when the services are up - some sort of health check, although not all services provide one. We will have to see if we can reuse what will be implemented in LocalEGA for testing here - for now this sleep is ok.

Comment (jbygdell) [reply .travis.yml:89]: If you look at the lines below that is what they do. In the sense that we make sure that everything is up and running.

**Review by blankdots — APPROVED** [review]
We might need to add in the `prerequisites` the fact that we recommend `kubernetes  1.13.0 +`. 
There are some comments for future PRs but other than that I approve.

Small note, we will need a `testing` phase in travis to test the functionality, but a future PR, we should make issues about these.

File: .travis.yml (hunk 2 above)
Comment (blankdots) [reply .travis.yml:89]: true, then we don't need to sleep here ?

Comment (jbygdell) [reply .travis.yml:89]: The sleep is to not spam messages until the majority of the services has had time to start, but it might be removed in the future.

**Comment by jonandernovella** [comment]
@jbygdell @blankdots Thanks, good job guys!

File: .travis.yml (hunk 2 above)
Comment (blankdots) [reply .travis.yml:89]: we could have a background script that does all those checks, and displays something when everything is up.

**Pull request merged**